### PR TITLE
initial support for timestamps, introduced delays

### PR DESCRIPTION
**Pull request by amygdala**

### CODE REVIEW

File: gce-cmdline-publisher/traffic_pubsub_generator.py [hunk 1]
```diff
@@ -94,17 +94,79 @@ def create_pubsub_client():
     return discovery.build('pubsub', 'v1beta2', http=http)
 
 
-def publish(client, pubsub_topic, data_line):
+def publish(client, pubsub_topic, data_line, msg_attributes=None):
```
Comment (tmatsuo) [inline gce-cmdline-publisher/traffic_pubsub_generator.py]: I'd rather make msg_attributes a dictionary, which makes it more generic?

File: gce-cmdline-publisher/traffic_pubsub_generator.py [hunk 2]
```diff
@@ -94,17 +94,79 @@ def create_pubsub_client():
...
     """Publish to the given pubsub topic."""
     pub = base64.urlsafe_b64encode(data_line)
```
Comment (tmatsuo) [inline gce-cmdline-publisher/traffic_pubsub_generator.py]: It's not in this change, but the name `pub` seems weird. How about `data`?

File: gce-cmdline-publisher/traffic_pubsub_generator.py [hunk 3]
```diff
@@ -94,17 +94,79 @@ def create_pubsub_client():
...
+    """Called when not using --current flag; retaining original time
+    information in data."""
+    ms_delay = 600000  # 10 mins in ms
+    epoch = datetime.datetime(1970, 1, 1)
+    ts_int = int(
+        ((orig_date - epoch).total_seconds() * 1000) +
+        orig_date.microsecond/1000)
+    # 'random_delays' indicates whether to include random apparent delays
+    # in published data
+    if random_delays and random.random() < .005:
```
Comment (tmatsuo) [inline gce-cmdline-publisher/traffic_pubsub_generator.py]: Can you refactor the code related to random_delays?

**Comment by tmatsuo** [comment]
I reviewed it and put the first batch of comments.

File: gce-cmdline-publisher/traffic_pubsub_generator.py (hunk 2 above)
Comment (amygdala) [inline gce-cmdline-publisher/traffic_pubsub_generator.py]: Done.

File: gce-cmdline-publisher/traffic_pubsub_generator.py (hunk 3 above)
Comment (amygdala) [inline gce-cmdline-publisher/traffic_pubsub_generator.py]: Done.

**Comment by tmatsuo** [comment]
@amygdala Can you fix the travis build?

**Comment by amygdala** [comment]
done, sorry!  forgot to run pep8.  Amazing how many pep complaints I can generate with just a few new lines.

**Comment by tmatsuo** [comment]
LGTM

**Pull request merged**